File: 0426-38-2523-closest-prime-numbers-in-range/0426-38-2523-closest-prime-numbers-in-range.java
import java.util.Arrays;

class Solution {
    boolean[] vis;

    public int[] closestPrimes(int left, int right) {
        sieve(right);
        int a = -1, b = -1;
        int lst = -1;
        int minGap = Integer.MAX_VALUE;
        for (int i = left; i <= right; i++) {
            if (vis[i]) continue;
            if (lst != -1 && i - lst < minGap) {
                minGap = i - lst;
                a = lst;
                b = i;
            }
            lst = i;
        }
        return new int[]{a, b};
    }

    private void sieve(int n) {
        vis = new boolean[n + 1];
        vis[0] = vis[1] = true;
        for (int i = 2; i * i <= n; i++) {
            if (!vis[i]) {
                for (int j = i + i; j <= n; j += i) {
                    vis[j] = true;
                }
            }
        }
    }
}

File: 0426-38-2523-closest-prime-numbers-in-range/0426-38-2523-closest-prime-numbers-in-range.js
/**
 * @param {number} left
 * @param {number} right
 * @return {number[]}
 */
var closestPrimes = function(left, right) {
        const vis = new Array(right + 1).fill(false);
        vis[0] = vis[1] = true;
        for (let i = 2; i * i <= right; i++) {
            if (!vis[i]) {
                for (let j = i + i; j <= right; j += i) {
                    vis[j] = true;
                }
            }
        }

        let a = -1, b = -1;
        let lst = -1;
        let minGap = Infinity;
        for (let i = left; i <= right; i++) {
            if (vis[i]) continue;
            if (lst !== -1 && i - lst < minGap) {
                minGap = i - lst;
                a = lst;
                b = i;
            }
            lst = i;
        }
        return [a, b];
    

};